addon/states.js

```diff
@@ -168,9 +169,9 @@ export default FragmentRootState;
 
 export function fragmentDidDirty(record, key, fragment) {
   if (!get(record, 'isDeleted')) {
-    // Add the fragment as a placeholder in the owner record's
+    // Add the fragment data as a placeholder in the owner record's
     // `_attributes` hash to indicate it is dirty
-    record._internalModel._attributes[key] = fragment;
+    record._internalModel._attributes[key] = fragment && assign(assign({}, fragment._data), fragment._attributes);
```

Comment:
Assign takes multiple sources, like so: `assign({}, fragment._data, fragment._attributes)`. Any reason to not do that?